internal/generator/generator.go

```diff
@@ -11,10 +11,33 @@ import (
 	"io"
 )
 
-// Generate receives a string sourcePath and an io.Writer writer. It will
-// extract all the directives definitions from the .c and .cpp files in
-// sourcePath and its subdirectories, then output the corresponding directive
-// masks map named "directives" and matchFunc named "Match" via writer.
-func Generate(sourcePath string, writer io.Writer, filter map[string]struct{}, override map[string][]Mask) error {
-	return genFromSrcCode(sourcePath, "directives", "Match", writer, filter, override)
+type GenerateConfig struct {
+	Filter map[string]struct{}
+
+	Override map[string][]Mask
```

Comment:
Nit: I think these could use comments too to at least explain what `string` is.

Reply:
Oh sure! These are useful information to add. Added in latest commit